crates/core/src/db/datastore/system_tables.rs

```diff
@@ -33,6 +33,21 @@ pub(crate) const ST_INDEXES_NAME: &str = "st_indexes";
 pub(crate) const ST_CONSTRAINTS_NAME: &str = "st_constraints";
 pub(crate) const ST_MODULE_NAME: &str = "st_module";
 
+/// Reserved range of sequence values used for system tables.
+///
+/// Ids for user-created tables will start at `ST_RESERVED_SEQUENCE_RANGE + 1`.
+///
+/// The range applies to all sequences allocated by system tables, i.e. table-,
+/// sequence-, index-, and constraint-ids.
+/// > Note that column-ids are positional indices and not based on a sequence.
+///
+/// These ids can be referred to statically even for system tables introduced
+/// after a database was created, so as long as the range is not exceeded.
+///
+/// However unlikely it may seem, it is advisable to check for overflow in the
+/// test suite when adding sequences to system tables.
+pub(crate) const ST_RESERVED_SEQUENCE_RANGE: u32 = 4096;
```

Comment:
I wonder if this should be equal to `SEQUENCE_PREALLOCATION_AMOUNT` (sadly defined in the `sats` crate).

I have deliberately used `ST_RESERVED_SEQUENCE_RANGE` to set the `allocated` value of system sequence, so changing the former should not break anything.